src/light_data/light_store.rs

```diff
@@ -107,7 +107,11 @@ impl<F: LurkField> LightStore<F> {
 
         // TODO: this needs to bail on encountering an opaque pointer
         while let Some(LightExpr::SymCons(s, ss)) = self.get(&ptr).flatten() {
-            let string = self.insert_scalar_string(s, store)?;
+            let string = if s == ScalarPtr::from_parts(ExprTag::Str.as_field(), F::zero()) {
```

Comment:
I would rather take the definition of `strnil_ptr` in `insert_scalar_string` (as well as `symnil_ptr`) here, export that to a `const` of the present struct, and reuse here. 
This may require making `ScalarPtr::from_parts` a const function.

Reply:
THis is a bit more involved than what I suggested, never mind.